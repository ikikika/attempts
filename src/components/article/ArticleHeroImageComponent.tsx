import React from "react";
import Image from "react-bootstrap/Image";
import parse from "html-react-parser";
import MediaImageType from "../../types/MediaImageType";

interface Props {
  mediaImageData: MediaImageType;
}

const styles = {
  byline: {
    background: "var(--c-image-byline-bg)",
    color: "var(--c-white-color)",
    fontSize: "var(--fs-byline)",
    fontWeight: 400,
    fontFamily: "var(--ff-byline)",
    lineHeight: "var(--lh-byline)",
  },
  caption: {
    fontSize: "var(--fs-caption-l)",
    lineHeight: "var(--lh-caption-l)",
  },
};

const ArticleHeroImageComponent = ({ mediaImageData }: Props) => {
  return (
    <>
      <div className="position-relative">
        <Image
          src={mediaImageData.src}
          alt={mediaImageData.byline}
          className="w-100"
        />
        <div
          className="position-absolute bottom-0 end-0 p-2"
          style={styles.byline}
        >
          {mediaImageData.byline}
        </div>
      </div>
      <div className="mt-2" style={styles.caption}>
        {parse(mediaImageData.description)}
      </div>
    </>
  );
};

export default ArticleHeroImageComponent;
