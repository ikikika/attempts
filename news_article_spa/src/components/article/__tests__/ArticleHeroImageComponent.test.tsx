import React from "react";
import renderer from "react-test-renderer";

import ArticleHeroImageComponent from "../ArticleHeroImageComponent";

it("ArticleHeroImageComponent matches snapshot", () => {
  const tree = renderer
    .create(
      <ArticleHeroImageComponent
        mediaImageData={{
          src: "test.png",
          byline: "lorem",
          description: "ipsum",
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
