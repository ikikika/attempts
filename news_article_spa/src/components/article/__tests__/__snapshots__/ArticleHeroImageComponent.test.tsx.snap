// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleHeroImageComponent matches snapshot 1`] = `
Array [
  <div
    className="position-relative"
  >
    <img
      alt="lorem"
      className="w-100"
      src="test.png"
    />
    <div
      className="position-absolute bottom-0 end-0 p-2"
      style={
        Object {
          "background": "var(--c-image-byline-bg)",
          "color": "var(--c-white-color)",
          "fontFamily": "var(--ff-byline)",
          "fontSize": "var(--fs-byline)",
          "fontWeight": 400,
          "lineHeight": "var(--lh-byline)",
        }
      }
    >
      lorem
    </div>
  </div>,
  <div
    className="mt-2"
    style={
      Object {
        "fontSize": "var(--fs-caption-l)",
        "lineHeight": "var(--lh-caption-l)",
      }
    }
  >
    ipsum
  </div>,
]
`;
